File: tests/sample.py
# sample.py — visual eyeball buffer.
# Use `python-ts-mode' with `treesit-font-lock-level' = 4.

from __future__ import annotations

from dataclasses import dataclass
from typing import Iterable


@dataclass(frozen=True)
class Rectangle:
    """A simple rectangle, doubling as a syntax-coverage sample."""

    width: int = 1
    height: int = 1

    @property
    def area(self) -> int:
        return self.width * self.height

    def scale(self, factor: float) -> "Rectangle":
        return Rectangle(int(self.width * factor), int(self.height * factor))


def total_area(rects: Iterable[Rectangle]) -> int:
    # Comprehensions, builtins, numbers, operators, f-strings:
    return sum(r.area for r in rects if r.area > 0)


if __name__ == "__main__":
    rects = [Rectangle(15, 7), Rectangle().scale(2.5)]
    for i, r in enumerate(rects, start=1):
        print(f"#{i}: {r.width}x{r.height} = {r.area}")
    print(f"total = {total_area(rects)}")

File: tests/sample.java
// sample.java — visual eyeball buffer for kanagawa-dragon-nvim.
// Open this file with `M-x java-ts-mode' (or rely on file-local
// auto-mode) and `(setq treesit-font-lock-level 4)' to reproduce the
// high-granularity highlighting the screenshot was taken from.

public class Rectangle {
    int width;
    int height;

    Rectangle() {
        this.width = 1;
        this.height = 1;
    }

    Rectangle(int width, int height) {
        this.width = width;
        this.height = height;
    }

    void calculateArea() {
        System.out.println(width + "x" + height + " = " + width * height);
    }

    void setWidth(int width) {
        this.width = width;
    }

    void setHeight(int height) {
        this.height = height;
    }

    public static void main(String[] args) {
        Rectangle rec1 = new Rectangle();
        Rectangle rec2 = new Rectangle(15, 7);

        System.out.println("Rectangle 1:");
        rec1.calculateArea();
        System.out.println("Rectangle 2:");
        rec2.calculateArea();

        rec2.setHeight(23);
        rec2.calculateArea();
    }
}
